cabal-install/Distribution/Solver/Modular/Explore.hs

```diff
@@ -60,7 +60,10 @@ backjump (EnableBackjumping enableBj) var initial xs =
           | otherwise                            = f (csAcc `CS.union` cs) cm'
 
     logBackjump :: ConflictSet QPN -> ConflictMap -> ConflictSetLog a
-    logBackjump cs cm = failWith (Failure cs Backjump) (cs, cm)
+    logBackjump cs cm = failWith (Failure cs Backjump) (cs, updateCM initial cm)
```

Comment:
We should benchmark `updateCM initial cm` versus `updateCM cs cm`. At least in the examples I tried, the latter gives even better results in terms of performance. On the other hand, @kosmikus has reservations that using `cs` here leads to double-counting some conflicts.